R/format_PEW_temp.R

```diff
@@ -0,0 +1,541 @@
+                  # there is only DateEgg1.
+                  # In metadata, there is a note that second and other broods are not recorded,
+                  # so can we assume that each brood is the first and only one?
+                  # BroodIDLaid = ifelse(RingAge == "chick",
+                  #                      paste(BreedingSeason, LocationID,
+                  #                            paste0(lubridate::year(LaidDate),
+                  #                                   sprintf("%02d", lubridate::month(LaidDate)),
+                  #                                   sprintf("%02d", lubridate::day(LaidDate))),
+                  #                            sep = "_"),
+                  #                      NA_character_)
+                  BroodIDLaid    = ifelse(RingAge == "chick",
+                                          paste(BreedingSeason, LocationID, sep = "_"),
+                                          NA_character_),
+                  BroodIDFledged = BroodIDLaid) %>%
+    dplyr::group_by(IndvID) %>%
+    dplyr::mutate(Sex_calculated = purrr::map_chr(.x = list(unique(na.omit(Sex_observed))), .f = ~{
+                                                            if(length(..1) == 0){
+                                                              return(NA_character_)
+                                                            } else if(length(..1) == 1){
+                                                              return(..1)
+                                                            } else {
+                                                              return("C")
+                                                            }
+                                                          }),
+                  Sex_genetic = NA_character_) %>%
+    dplyr::arrange(CaptureDate) %>%
+    dplyr::summarize(Species = first(Species),
+                     PopID = "PEW",
+                     IndvID = first(IndvID),
+                     RingSeason = first(BreedingSeason),
+                     RingAge = first(RingAge),
+                     BroodIDLaid = first(BroodIDLaid),
+                     BroodIDFledged = first(BroodIDFledged),
+                     Sex_calculated = first(Sex_calculated),
+                     Sex_genetic = first(Sex_genetic)) %>%
+    dplyr::ungroup() %>%
+    #### Final arrangement
+    dplyr::select(IndvID, Species, PopID, BroodIDLaid, BroodIDFledged,
+                  RingSeason, RingAge, Sex_calculated, Sex_genetic)
+
+  return(Individual_data)
+
+  }
+
+create_location_PEW <- function(data) {
+
+  Location_data <-
+    data %>%
+    dplyr::select(BreedingSeason, Date, NestboxID, PopID) %>%
+    group_by(NestboxID) %>%
+    arrange(BreedingSeason, Date) %>%
+    dplyr::summarise(StartSeason = first(BreedingSeason),
+                     #### CHECK WITH DATA OWNER
+                     #### EndSeason = last(BreedingSeason))
```

Comment:
BreedingSeason is an integer, so you can just use min/max rather than arrange/first. arrange/first has the risk of failing if there is an NA Date, because these won't arrange properly.

```suggestion
    dplyr::summarise(StartSeason = min(BreedingSeason),
                     #### CHECK WITH DATA OWNER
                     #### EndSeason = max(BreedingSeason))
```

Reply:
Ok.